### PR TITLE
Create LED methods and access them by their colors.

### PR DESCRIPTION
**Pull request by RaulTrombin**
This PR introduces the led_set and led_toggle methods, as well as an enum list of available colors for selection. 
The changes are fully documented and supported by examples.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -187,6 +211,37 @@ impl Led {
         [&mut self.first, &mut self.second, &mut self.third]
     }
 
+    pub fn select_by_color(&mut self, color: LedColor) -> &mut Pin {
+        let pin: &mut Pin = match color {
+            LedColor::Green => &mut self.first,
+            LedColor::Blue => &mut self.second,
+            LedColor::Red => &mut self.third,
+        };
```
Comment (patrickelectric) [inline src/lib.rs:219]: you can return directly from the match

File: src/lib.rs [hunk 2]
```diff
@@ -187,6 +211,37 @@ impl Led {
...
+        pin
+    }
+
+    pub fn set_led(&mut self, color: LedColor, state: bool) {
+        let state_u8 = if state { 0 } else { 1 }; // LEDs are enabled with logic low
```
Comment (patrickelectric) [inline src/lib.rs:224]: this is unnecessary, you can do `state.into()`

File: src/lib.rs [hunk 3]
```diff
@@ -505,11 +560,91 @@ impl Navigator {
...
+    ///     nav.set_led(LedColor::Green, false);
+    ///     sleep(Duration::from_millis(1000));
+    ///     nav.set_led(LedColor::Blue, true);
+    ///     sleep(Duration::from_millis(1000));
+    ///     nav.set_led(LedColor::Blue, false);
+    ///     sleep(Duration::from_millis(1000));
+    ///     nav.set_led(LedColor::Red, true);
+    ///     sleep(Duration::from_millis(1000));
+    ///     nav.set_led(LedColor::Red, false);
+    ///     sleep(Duration::from_millis(1000));
```
Comment (patrickelectric) [inline src/lib.rs:592]: that appears really unnecessary.. please make the examples simpler.

**Review by patrickelectric — CHANGES_REQUESTED** [review]

File: src/lib.rs (hunk 3 above)
Comment (RaulTrombin) [reply src/lib.rs:592]: it was blinking pretty well, but okay. will keep just green led.

File: src/lib.rs (hunk 2 above)
Comment (RaulTrombin) [reply src/lib.rs:224]: idk why it was not working before.

File: src/lib.rs [hunk 4]
```diff
@@ -187,6 +211,34 @@ impl Led {
...
+    }
+
+    pub fn led_toggle(&mut self, pin: LedColor) {
+        let mut state: bool = false;
+
+        let pin_struct = self.select_by_color(pin);
+
+        if pin_struct.get_value().unwrap() == 1 {
+            state = true
+        }
```
Comment (patrickelectric) [inline src/lib.rs:237]:
```suggestion
        let state = pin_struct.get_value().unwrap() == 1;
```

File: src/lib.rs [hunk 5]
```diff
@@ -187,6 +211,30 @@ impl Led {
...
+    }
+
+    pub fn led_toggle(&mut self, pin: LedColor) {
+        let pin_struct = self.select_by_color(pin);
+
+        let state = pin_struct.get_value().unwrap() == 1;
+
+        self.set_led(pin, state)
+    }
+
```
Comment (joaoantoniocardoso) [inline src/lib.rs:237]: Can we add a method to read the leds state?

**Comment by joaoantoniocardoso** [comment]
Can we add a method to read the leds state?

**Comment by RaulTrombin** [comment]
> Can we add a method to read the leds state?

Please take a look, added new commit that implements this method.

**Review by patrickelectric — APPROVED** [review]

**Pull request merged**